src/Service.Tests/GraphQLRequestExecutor.cs

```diff
@@ -61,5 +72,63 @@ internal static class GraphQLRequestExecutor
 
             return graphQLResult.GetProperty("data").GetProperty(queryName);
         }
+
+        /// <summary>
+        /// Executes a graphql query and returns the result as a json element.
+        /// Used for multiple datasources tests.
+        /// </summary>
+        /// <param name="client">client</param>
+        /// <param name="configProvider">configProvider</param>
+        /// <param name="query">query</param>
+        /// <param name="variables">variables</param>
+        /// <param name="authToken">authToken.</param>
+        /// <param name="clientRoleHeader">clientRoleHeader</param>
+        /// <returns>JsonResult</returns>
+        public static async Task<JsonElement> PostGraphQLRequestAsync(
+            HttpClient client,
+            RuntimeConfigProvider configProvider,
+            string query,
+            Dictionary<string, object> variables = null,
+            string authToken = null,
+            string clientRoleHeader = null)
+        {
+            object payload = variables == null ?
+                new { query } :
+                new
+                {
+                    query,
+                    variables
+                };
+
+            string graphQLEndpoint = configProvider.GetConfig().Runtime.GraphQL.Path;
+
+            HttpRequestMessage request = new(HttpMethod.Post, graphQLEndpoint)
+            {
+                Content = JsonContent.Create(payload)
+            };
+
+            if (!string.IsNullOrEmpty(authToken))
+            {
+                request.Headers.Add(AuthenticationOptions.CLIENT_PRINCIPAL_HEADER, authToken);
+            }
+
+            if (!string.IsNullOrEmpty(clientRoleHeader))
+            {
+                request.Headers.Add(AuthorizationResolver.CLIENT_ROLE_HEADER, clientRoleHeader);
+            }
+
+            HttpResponseMessage response = await client.SendAsync(request);
+            string body = await response.Content.ReadAsStringAsync();
+
+            JsonElement graphQLResult = JsonSerializer.Deserialize<JsonElement>(body);
+
+            if (graphQLResult.TryGetProperty("errors", out JsonElement errors))
+            {
+                // to validate expected errors and error message
+                return errors;
+            }
+
+            return graphQLResult.GetProperty("data");
```

Comment:
Except for the return value, everything else seems to be copied from the above function. Can we put the common code in a helper and reuse?

Reply:
Function not used. was added by error.